scripts/zprogbar.sh

```diff
@@ -0,0 +1,53 @@
+#!/bin/bash
+tput civis
+cols=$(tput cols)                                                                         
+lines=$(( $(tput lines) - 4 ))
+maxchars=$(($cols * $lines))
+denom=$(($maxchars / 100))
+max=$(( 100 * $denom ))
+sleep=.5
+
+sp="/-\¦"
+sc=0
+spin() {
+   printf "\b${sp:sc++:1}"
+   ((sc==${#sp})) && sc=0                                                                 
+}                                                                                         
+
+while [[ $percent -ne 100 ]]
+do
+
+text="$(zpool status $1)"
+percent=$( echo "$text" | awk '/% done/{$1 = "" ; $2 = "" ; $4 = "" ; $5 ="" ; $6 = "" ; $7 = "" ; $8 = "" ; $9 = "" ; print substr($0, 1, length($0)-7)}')
```

Comment:
one more tip, if you're only trying to get the stuff before % in a long sentence, you can:
```
leftside=${text%\%*}
percent=${leftside##* }
```
no need to call awk

Reply:
I'm still learning Bash and github, can people submit pull requests at the moment, because I really don't want to have to learn the style guides or maintain the script forever?